### PR TITLE
Make JRB mist and DDD objects act-specific.

### PR DESCRIPTION
**Pull request by Arceveti**
Moved from #235

### CODE REVIEW

**Review by Mr-Wiseguy — APPROVED** [review]
Very good options for vanilla level hacks to have

File: include/config/config_objects.h [hunk 1]
```diff
@@ -73,3 +73,13 @@
 
 // The number of segments Wiggler has, not including the head. Vanilla is 4.
 #define WIGGLER_NUM_SEGMENTS     4
+
+// -- JRB MIST --
+
+// Makes the JRB mist always appear on act 1.
+#define JRB_ACT_SPECIFIC_MIST
```
Comment (Mr-Wiseguy) [inline include/config/config_objects.h:80]: These new defines should probably be off by default

**Review by Mr-Wiseguy — CHANGES_REQUESTED** [review]
Oh actually the defines should be off by default

File: include/config/config_objects.h (hunk 1 above)
Comment (Arceveti) [reply include/config/config_objects.h]: Done

**Comment by arthurtilly** [comment]
these should at the very least be the same define, and honestly i'm not sure if i agree with them being a define at all because almost no-one likes how the vanilla implementation is and it's a very small thing. i'd much prefer if this was enabled by default, but otherwise at the very least it should be one define.

**Comment by gheskett** [comment]
I don't really get why this is a big enough deal to warrant change anyway. And even if it does, it's definitely not worthy of a config file define in my eyes.

**Comment by arthurtilly** [comment]
yep agreed, i think it should just be a quick change thats always enabled, i cant think of any scenario anyone would be against this even for a vanilla hack

**Review by gheskett — APPROVED** [review]
I'm still a little on the fence with this PR but I don't personally care about it enough to shut it down.

**Review by arthurtilly — APPROVED** [review]
approved

**Comment by gheskett** [comment]
Wait hang on, so deleting Bowser sub double whammies as a means of opening the Bowser sub door, but how does the current that sucks Mario into the doorway function? Is it always enabled no matter what or does it rely on the same thing as the sub originally did?

**Review by gheskett — CHANGES_REQUESTED** [review]
Actually yeah the currents/whirlpools are still dependent on Bowser 2. This might be as simple as changing this line:
`WHIRLPOOL(/*unk2*/ 1, /*unk3*/ 2, /*pos*/ 3917, -2040, -6041, /*strength*/ 50),`
unk3 should be set to 3.

Alternatively (and preferably), the level command should just be cleaned up because it's honestly pretty useless.

**Comment by arthurtilly** [comment]
hurry up arceveti

**Comment by Reonu** [comment]
@Arceveti pls

**Comment by Arceveti** [comment]
> Actually yeah the currents/whirlpools are still dependent on Bowser 2. This might be as simple as changing this line: `WHIRLPOOL(/*unk2*/ 1, /*unk3*/ 2, /*pos*/ 3917, -2040, -6041, /*strength*/ 50),` unk3 should be set to 3.
> 
> Alternatively (and preferably), the level command should just be cleaned up because it's honestly pretty useless.

Whirlpools are fixed now and I cleaned up the level command a bit, now it uses acts similar to `OBJECT_WITH_ACTS` instead of specific hardcoded conditions.

**Comment by Arceveti** [comment]
Also fixes #366 now

**Review by gheskett — APPROVED** [review]
[d8daaf6](https://github.com/HackerN64/HackerSM64/commit/d8daaf63437a934a745281186a586067d6125913) comments aside, LGTM

**Pull request merged**